src/movescount/movescount.cpp

```diff
@@ -574,16 +575,22 @@ void MovesCount::writeLogInThread(LogEntry *logEntry)
 
     reply = syncPOST("/moves/", "", output, true);
 
-    if (reply->error() == QNetworkReply::NoError || reply->error() == QNetworkReply::ContentConflictError) {
-        QByteArray data = reply->readAll();
+    if (reply->error() == QNetworkReply::NoError){
+        data = reply->readAll();
         if (jsonParser.parseLogReply(data, moveId) == 0) {
             emit logMoveID(logEntry->device, logEntry->time, moveId);
         } else {
             qDebug() << "Failed to upload log, movescount.com replied with \"" << reply->readAll() << "\"";
+            emit uploadError(data);
         }
     } 
+    else if(reply->error() == QNetworkReply::ContentConflictError){
+        emit uploadError("Move already uploaded");
+    }
     else {
+        data = reply->readAll();
         qDebug() << "Failed to upload log (err code:" << reply->error() << "), movescount.com replied with \"" << reply->readAll() << "\"";
+        emit uploadError(data);
```

Comment:
I probably don't understand the code fully here, but doesn't that mean we always show the message-box, also when no error occured during uploading?

Reply:
It's only uploaded if there're no errors:
> if (jsonParser.parseLogReply(data, moveId) == 0) {

Actually, it can be simplified or rewrite it, but I wanted to distinguish REST API field errors, duplicated moves and JSON parser errors.

Reply:
ah, yes, now I see, the "good" path is in the middle, than it will be fine for now, we can adjust it in a 2nd step.